### PR TITLE
Init

### PR DESCRIPTION
**Pull request by smiller171**
Use `rhythmictech/terraform-terraform-find-semver-match` to find the appropriate version from Github Releases.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -1,16 +1,32 @@
...
 }
 
 locals {
-  # tflint-ignore: terraform_unused_declarations
-  name = module.tags.name
-  # tflint-ignore: terraform_unused_declarations
-  tags = module.tags.tags_no_name
+  data               = jsondecode(data.http.available_versions.body)
+  available_versions = local.data.*.tag_name
+  tag_version_map = {
```
Comment (cdaniluk) [inline main.tf:13]:
```suggestion

    tag_version_map = {
```

File: main.tf [hunk 2]
```diff
@@ -1,16 +1,32 @@
...
+    accept = "vnd.github.v3+json"
+  }
 }
 
 locals {
-  # tflint-ignore: terraform_unused_declarations
-  name = module.tags.name
-  # tflint-ignore: terraform_unused_declarations
-  tags = module.tags.tags_no_name
+  data               = jsondecode(data.http.available_versions.body)
```
Comment (cdaniluk) [inline main.tf]: sort

File: variables.tf [hunk 3]
```diff
@@ -1,11 +1,15 @@
 
-variable "name" {
-  description = "Moniker to apply to all resources in the module"
+variable "repo_name" {
   type        = string
```
Comment (cdaniluk) [inline variables.tf:7]: sort desc first throughout this file

File: versions.tf [hunk 4]
```diff
@@ -1,3 +1,13 @@
 terraform {
   required_version = ">= 0.12.14"
```
Comment (cdaniluk) [inline versions.tf]:
```suggestion
  required_version = ">= 0.12.19"
```

just to be consistent with what we've been enforcing in general.

File: main.tf [hunk 5]
```diff
@@ -1,16 +1,30 @@
...
+module "find_semver_match" {
+  source  = "rhythmictech/find-semver-match/terraform"
+  version = "1.0.0-rc3"
+
+  available_versions = local.available_versions
+  version_constraint = var.version_constraint
+}
+
+data "github_release" "this" {
+  repository  = var.repo_name
```
Comment (cdaniluk) [inline main.tf]: sort

**Review by cdaniluk — APPROVED** [review]
Requested some syntactical things but overall love this.

**Review by sblack4 — APPROVED** [review]

**Pull request merged**